forum/qa-plugin/tips-widget/tips-widget-layer.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+class qa_html_theme_layer extends qa_html_theme_base
+{
+	function initialize()
+	{
+		if (!qa_opt('tips-enable'))
+		{
+			qa_html_theme_base::initialize();
+			return;
+		}
+		
+		if ($widget_module = qa_load_module('widget', 'Tips Widget Widget'))
```

Comment:
nigdy, przenigdy nie tworzymy nowego stanu w warunku. Taki zapis nigdy nie powinien się pojawić w kodzie.

Reply:
Dlaczego? Nie widzę powodu, aby tak nie robić.

Reply:
1) jest to nieczytelne 2) jest to ryzykowne 3) nie po to jest instrukcja warunkowa aby miała dodatkową odpowiedzialność za błędy wykonania funkcji 'qa_load_module' - wystarczy, że rzuci wyjątkiem lub zwróci jakikolwiek typ warninga, a ten if i tak się wykona.

Reply:
Ta funkcja przy niepowodzeniu zwraca nulla. Po drugie ten warunek nie wykrywa błędu, lecz jest częścią poprawnego przebiegu kodu, gdyż jeśli admin nie włączy wtyczki widgetu, to właśnie ten warunek ma uniemożliwić ustawienie nowego ciasteczka.

Reply:
A dajmy na to, że będzie jakaś sytuacja, która jednak wywali exception? Co wtedy? Nawet jeśli ma jakieś uzasadnienie, że nie jest to groźne dla tej funkcji, to i tak stosowanie takiego zapisu jest po prostu generującym przyszłe problemy trudne do zdiagnozowania.

Reply:
Autorzy Q2A deklarują, że starsze funkcje będą ciągle wspierane w takiej formie, jaka aktualnie jest. Nie widzę więc powodu negowania tego zapisu, gdyż istnieje prawie zerowe prawdopodobieństwo, że ktoś z deweloperów tego frameworku przez przypadek zapomni o tym i będzie zwracać z tej funkcji wyjątki. Równie dobrze mógłbym zabezpieczyć kod przez innymi nierealnymi przypadkami, ale chyba poprawnym jest założenie pewnych stanów, które biblioteka oferuje, bez wchodzenia w szczegóły.

Reply:
Co i tak nie zmienia tego, że jest to zapis niepoprawny. Nawet w standardach C++ odradza się przypisywania wartości w instrukcji warunkowej. To działa. Ale nie daje nikomu żadnej pewności odnośnie skutków, których raczej nie przewidzisz.